public/app/features/dashboard/dashgrid/PanelHeader/PanelHeader.tsx

```diff
@@ -1,175 +1,50 @@
-  };
-
-  isClick = (clickCoordinates: ClickCoordinates) => {
-    return clickCoordinates.x === this.clickCoordinates.x && clickCoordinates.y === this.clickCoordinates.y;
-  };
-
-  onMenuToggle = (event: React.MouseEvent<HTMLDivElement>) => {
-    if (!this.isClick(this.eventToClickCoordinates(event))) {
-      return;
-    }
-
-    event.stopPropagation();
-
-    const { dashboard, panel, angularComponent } = this.props;
-    const menuItems = getPanelMenu(dashboard, panel, angularComponent);
-
-    this.setState({
-      panelMenuOpen: !this.state.panelMenuOpen,
-      menuItems,
-    });
-  };
-
-  closeMenu = () => {
-    this.setState({
-      panelMenuOpen: false,
-    });
-  };
-
+export class PanelHeader extends PureComponent<Props> {
   onCancelQuery = () => {
     this.props.panel.getQueryRunner().cancelQuery();
   };
 
-  renderLoadingState(state: LoadingState): JSX.Element | null {
-    if (state === LoadingState.Loading) {
-      return (
-        <div className="panel-loading" onClick={this.onCancelQuery}>
-          <Tooltip content="Cancel query">
-            <Icon className="panel-loading__spinner spin-clockwise" name="sync" />
-          </Tooltip>
-        </div>
-      );
-    }
-
-    if (state === LoadingState.Streaming) {
-      const styles = getStyles();
-
-      return (
-        <div className="panel-loading" onClick={this.onCancelQuery}>
-          <div title="Streaming (click to stop)" className={styles.streamIndicator} />
-        </div>
-      );
-    }
-
-    return null;
-  }
-
-  openInspect = (e: React.SyntheticEvent, tab: string) => {
-    const { updateLocation, panel } = this.props;
-
-    e.stopPropagation();
-
-    updateLocation({
-      query: { inspect: panel.id, inspectTab: tab },
-      partial: true,
-    });
-  };
-
-  // This will show one icon for each severity
-  renderNotice = (notice: QueryResultMetaNotice) => {
-    let iconName: IconName = 'info-circle';
-    if (notice.severity === 'error' || notice.severity === 'warning') {
-      iconName = 'exclamation-triangle';
-    }
-
-    return (
-      <Tooltip content={notice.text} key={notice.severity}>
-        {notice.inspect ? (
-          <div className="panel-info-notice pointer" onClick={(e) => this.openInspect(e, notice.inspect!)}>
-            <Icon name={iconName} style={{ marginRight: '8px' }} />
-          </div>
-        ) : (
-          <a className="panel-info-notice" href={notice.link} target="_blank" rel="noreferrer">
-            <Icon name={iconName} style={{ marginRight: '8px' }} />
-          </a>
-        )}
-      </Tooltip>
-    );
-  };
-
   render() {
-    const { panel, error, isViewing, isEditing, data, alertState } = this.props;
-    const { menuItems } = this.state;
+    const { panel, error, isViewing, isEditing, data, alertState, dashboard } = this.props;
     const title = panel.replaceVariables(panel.title, {}, 'text');
 
     const panelHeaderClass = classNames({
```

Comment:
Since we are here, maybe let's get rid of `classNames` in favour of `cx`?

Reply:
Sure thing!

Reply:
Fixed!